File: Frontend/src/app/models/computer_ComputerPart.ts
import { Computer } from "./computer";
import { ComputerPart } from "./computerPart";

/*
*This is the object structure of the computer_ComputerPart object.
*It is used to store it in the database from frontend to backend.
*/

export interface Computer_ComputerPart {
    id: number, 
    computerId: number,
    computerPartId: number,
    // *get the computer object from the computer.ts file
    computer?: Computer,
    // *get the computerPart object from the computerPart.ts file
    computerPart?: ComputerPart
}

File: Backend/SOP/SOP/Entities/Computer_ComputerPart.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SOP.Entities
{
    public class Computer_ComputerPart
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Computer.Id")]
        public int ComputerId { get; set; }

        [ForeignKey("ComputerPart.Id")]
        public int ComputerPartId { get; set; }

        public Computer Computer { get; set; }

        public ComputerPart ComputerPart { get; set; }
    }
}
